tasks/libs/github_actions_tools.py

```diff
@@ -120,14 +122,71 @@ def follow_workflow_run(run):
         sleep(60)
 
 
-def print_workflow_conclusion(conclusion):
+def print_workflow_conclusion(conclusion, workflow_uri):
     """
     Print the workflow conclusion
     """
     if conclusion == "success":
         print(color_message("Workflow run succeeded", "green"))
     else:
         print(color_message(f"Workflow run ended with state: {conclusion}", "red"))
+        print(f"Failed workflow URI {workflow_uri}")
+
+
+def print_failed_jobs_logs(run):
+    """
+    Retrieves, parses and print failed job logs for the workflow run
+    """
+    failed_jobs = get_failed_jobs(run)
+
+    download_with_retry(download_logs, run, destination="workflow_logs")
+
+    failed_steps = get_failed_steps_log_files("workflow_logs", failed_jobs)
+
+    for failed_step, log_file in failed_steps.items():
+        print(color_message(f"Step: {failed_step} failed:", "red"))
+        print("\n".join(parse_log_file(log_file)))
+    print(color_message("Logs might be incomplete, for complete logs check directly in the worklow logs", "blue"))
+
+
+def get_failed_jobs(run):
+    """
+    Retrieves failed jobs for the workflow run
+    """
+    failed_jobs = []
+    jobs = run.jobs()
+    for job in jobs:
+        if job.conclusion == "failure":
+            failed_jobs.append(job)
+    return failed_jobs
```

Comment:
pythonic way of doing this method
```suggestion
    return [job for job in run.jobs() if job.conclusion == "failure"]
```